### ISSUE TITLE
megacmd on synology sync/get issues

### ISSUE THREAD

**Issue by nnachean** [meganz/MEGAcmd#521]
Okay, I've figured out most of my errors. They were mostly my fault surprise, surprise. Now I'm stuck with my get and sync commands. With the get command about 6 out of 50 folders populate before the mega-cmd-server process status goes to sleep. I thought it was because my hard drive went into hibernation but I turned it off and it still happened. So instead I used the sync command. It was going well until it came time for it to sync actual files. All of the folders populated but none of the files did. I found a closed issue from early 2019 that had a similar problem using the transfers command to cancel the jobs and get them back up and running but the transfers command has never worked for me. Nothing happens for a while. No error, nothing. I followed the Qnap instructions and added the mega commands to a path when I didn't have to, would that make it slow/unresponsive. Whenever mega-cmd is executing commands it's always extremely slow and gives me no indication that it's doing anything. I wait like 5 or 6 hours and it just comes up with my email to start another command.

like this: 

> sample@email.com :/$ sync /folder /folder
sample@email.com :/$ 

### tldr; get command only populates 6/50 folders, mega-cmd process goes into sleeping, it's not hdd hibernation. sync command populates all folders but no files. mega-cmd process running. other commands like transfers and sync take ages to execute and give no indication that they did execute until mega-cmd returns with the start of the command line.

- 
-
-
-
-
-
-
-
-

I sent this to support and they sent me here. I promise I really did try to figure this out on my own with my limited knowledge.

"I've been having issues just trying to login, let alone set up a sync. When I login, I type in my authentication code, and then it never moves past fetching nodes even though it reaches 100%. I've tried it twice. The first time I left it overnight, turned off my computer then came back to it and I was finally logged in. I tried downloading a folder from my mega, all it did was download some folders within that folder but no files and it never completed. I tried it again, nothing happened and I couldn't type in any commands, so I started another session. I typed in the mega command to get it started again but all it would show me is the megacmd title and welcome script. I couldn't put in any commands. I left it for a day again and was finally able to input commands again. Instead of trying to download again, I tried to set up a sync. I tried setting to a folder in my mega directly but it said it wasn't found(I tried a couple of variations of that folder's name) so I put that folder into another folder that I named with a single word. That caused megacmd to pause for about an hour and a half only to come up with an error again. So, I set up another session. I logged out but it wasn't doing anything so I left it overnight again, this time I left my pc on. I started a new session (these sessions are with putty btw not new sessions in mega) my email still popped up so I uninstalled it, removed any session that might still be connected on mega the website, and restarted my nas. That would be the second time I restarted it to get mega working again. I started a new session in putty, put the mega command and it works right away. Until I try to login again. It's still on fetching nodes even though it's on 100%. Nothing is happening. Am I doing something wrong? I'm using putty to connect via ssh. My nas is brand new with plenty of storage and there is nothing on it. The model is a DS220j if that helps. I've tried googling what I could and using the user guide and a step by step you have for QNAP. I must be doing something wrong. Is there a step I'm missing?"

I tried to recreate any of the errors but...
"It didn't even get to the fetching nodes this time. I had just closed a session to open the one in the picture and this is what it gave me. It's been like this for at least an hour. For me to recreate more I would have to be logged in and it's not doing anything, so this is all I have."

![Screenshot 2021-05-02 200557](https://user-images.githubusercontent.com/83561255/116833536-bde88900-ab87-11eb-9ca1-c6576590f77e.png)

I feel I'm doing something/missing something really stupid.

**Comment by nnachean** [comment]
Okay I found the megacmdserver log and these errors keep popping up:

> [info: 17:09:49] Login complete as sample@email.com
[err: 17:09:50] ERROR writing output Code to socket: 32
[err: 17:09:50] ERROR writing to socket: 32
[info: 22:19:59] Added sync: 
[err: 04:44:20] ERROR writing output Code to socket: 32
[err: 04:44:20] ERROR writing to socket: 32
[err: 05:22:49] ERROR writing output Code to socket: 32
[err: 05:22:49] ERROR writing to socket: 32

What is this error mean, what is it caused by and how can I fix it? This error seems to always pop up when I use any command that have to access my nas. (like the get, sync, and transfer commands) As well as the login command.

**Comment by nnachean** [comment]
Even with a factory reset megacmd comes up with these errors.

**Comment by RickDB** [comment]
Same issues here even after reset.
Was working fine for almost a year and version wise nothing changed, did try 1.4.0 to rule out compatibility issues however same results.

**Comment by saridzawa** [comment]
same issue

**Comment by polmr** [comment]
Newer releases of MEGAcmd for NAS (Synology/QNAP) featuring several improvements and fixes, can now be downloaded from here: https://mega.io/nas.
